packages/ckeditor5-coremedia-blocklist/src/blocklistui.ts

```diff
@@ -0,0 +1,443 @@
+import { Editor, Plugin } from "@ckeditor/ckeditor5-core";
+import { ifCommand } from "@coremedia/ckeditor5-core-common/src/Commands";
+import blocklistIcon from "../theme/icons/blacklist.svg";
+import { ButtonView, clickOutsideHandler, ContextualBalloon } from "@ckeditor/ckeditor5-ui";
+import {
+  TextProxy,
+  ViewAttributeElement,
+  ViewDocumentClickEvent,
+  ViewDocumentFragment,
+  ViewNode,
+  ViewPosition,
+} from "@ckeditor/ckeditor5-engine";
+import { PositionOptions } from "@ckeditor/ckeditor5-utils";
+import BlocklistCommand, { BLOCKLIST_COMMAND_NAME } from "./blocklistCommand";
+import BlocklistActionsView from "./ui/blocklistActionsView";
+import "./lang/blocklist";
+import { UnblockEvent } from "./ui/blockedWordView";
+import BlocklistEditing from "./blocklistediting";
+
+const BLOCKLIST_KEYSTROKE = "Ctrl+Shift+B";
+
+/**
+ * The Blocklist UI plugin. It introduces the `'blocklist'` button and support for the <kbd>Ctrl+Shift+B</kbd> keystroke.
+ * It also creates the views, that are displayed inside the contextual balloon when triggered by the button, click or
+ * keystroke.
+ *
+ * It uses the contextual balloon plugin}.
+ */
+export default class Blocklistui extends Plugin {
+  static readonly pluginName: string = "BlocklistUI";
+  static readonly requires = [ContextualBalloon, BlocklistEditing];
+
+  blocklistActionsView: BlocklistActionsView;
+
+  blocklistCommand: BlocklistCommand | undefined;
+
+  #balloon: ContextualBalloon | undefined = undefined;
+
+  constructor(editor: Editor) {
+    super(editor);
+    this.blocklistActionsView = this.#createBlocklistActionsView();
```

Comment:
Unsure: 
CKEditor started to initialize things lazy for performance reasons. I don't know if there is an impact right now, but we could consider to initialize it lazy too. Deeper in the class there is a check for existence anyway before using the field.